sdk/include/ds/linked_list.h

```diff
@@ -0,0 +1,642 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file A (circular) doubly linked list, abstracted over cons cell
+ * representations.
+ */
+
+#pragma once
+
+#include <concepts>
+#include <ds/pointer.h>
+
+namespace ds::linked_list
+{
+
+	namespace cell
+	{
+		/**
+		 * The primitive, required, abstract interface to our cons cells.
+		 *
+		 * All methods are "namespaced" with `cell_` to support the case where
+		 * the encoded forms are also representing other state (for example,
+		 * bit-packed flags in pointer address bits).
+		 */
+		template<typename T>
+		concept HasPrimOps = requires(T &t)
```

Comment:
Please don't use abbreviations in names (as per the style guide).  Does Prim stand for primitive?  Prime?  Per the comment, I presume primitive, though it's not really clear what 'primitive' means here.  Can it have a more descriptive name (e.g. `CellOperations`)?

Reply:
Done.